app/models/osf-model.ts

```diff
@@ -157,6 +158,80 @@ export default class OsfModel extends Model {
         return currentResults;
     }
 
+    async createM2MRelationship<
+    T extends OsfModel,
+    >(
+        this: T,
+        relationshipName: RelationshipsFor<T> & string,
+        relatedObjId: string,
+    ) {
+        const apiRelationshipName = underscore(relationshipName);
+        const url = this.relationshipLinks[apiRelationshipName].links!.self.href;
```

Comment:
This doesn't type-check now that I've fixed the types for relationship links. Will need to make sure the self link exists before accessing it.